core/hash_map.h

```diff
@@ -74,13 +78,14 @@ class HashMapHahserDefault {
  * @param TKey  Key, search is based on it, needs to be hasheable. It is unique in this container.
  * @param TData Data, data associated with the key
  * @param Hasher Hasher object, needs to provide a valid static hash function for TKey
+ * @param Comparator comparator object, needs to be able to safely compare two TKey values
```

Comment:
The comment you're referring to is for func f32hash(p unsafe.Pointer, h uintptr) uintptr; in Go. This is equivalent to my comment  '// Normalize +/- 0.0 and NaN values so they hash the same.' in hashfuncs.h.

Since in Go there is no way to use a value as a key that does not have an '==' operator there is never an issue. But the comment equivalent you're asking for here exists in my PR.